decidim-core/app/serializers/decidim/exporters/serializer.rb

```diff
@@ -19,12 +19,42 @@ def initialize(resource)
         @resource = resource
       end
 
+      # Publishes a serialize event and returns serialized hash by default (can be customized at the event).
+      def run
+        finalize(resource, serialize)
+      end
+
       # Public: Returns a serialized view of the provided resource.
       #
       # Returns a nested Hash with the fields.
       def serialize
         @resource.to_h
       end
+
+      # Public: Publishes notification (event) so that subscribers can modify serialized data.
+      #
+      # resource - The Object to serialize.
+      # serialized_data - Hash with the serialized data for this resource.
+      #
+      # Returns a nested Hash with the fields by default.
+      def finalize(resource, serialized_data)
```

Comment:
I think the resource is already always available within the object instance, so we don't necessarily need to pass it to this method?